### PR TITLE
fix: Core TokenMap to CSL multiasset insert de-dupe  

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
ADP-1763

# Proposed Solution
remove duplicated insert when assets from the same policy are present in the token map

# Important Info
Conversion back to core types test is failing. Currently investigating

### CODE REVIEW

File: packages/core/test/CSL/testData.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export const poolParameters: Cardano.PoolParameters = {
   vrf
 };
 
+// REVIEW: Also add these additional assets here?
```
Comment (mkazlauskas) [inline packages/core/test/CSL/testData.ts:33]: Good idea :+1:

File: packages/core/src/CSL/coreToCsl/coreToCsl.ts [hunk 2]
```diff
@@ -44,14 +44,29 @@ import { SerializationError } from '../../errors';
...
-    const amount = BigNum.from_str(assets.get(assetId)!.toString());
-    assetsObj.insert(assetName, amount);
-    multiasset.insert(scriptHash, assetsObj);
+  const policyObj: {
+    [policyId: string]: { assetsMap: Map<AssetName, BigNum>; scriptHash: ScriptHash };
+  } = {};
+
+  for (const assetId of map.keys()) {
+    const { assetName, scriptHash } = Asset.util.parseAssetId(assetId);
+    // REVIEW: Could also return the policyId from Asset.util.parseAssetId
```
Comment (mkazlauskas) [inline packages/core/src/CSL/coreToCsl/coreToCsl.ts:55]: It already returns it as `scriptHash` object. Perhaps it should be exported as `coreToCsl.asset` though :thinking:

File: packages/core/test/CSL/coreToCsl.test.ts [hunk 3]
```diff
@@ -40,15 +40,17 @@ describe('coreToCsl', () => {
...
-    for (const [assetId, expectedAssetQuantity] of txOut.value.assets!.entries()) {
-      const { scriptHash, assetName } = Asset.util.parseAssetId(assetId);
-      const assetQuantity = BigInt(multiasset.get(scriptHash)!.get(assetName)!.to_str());
-      expect(assetQuantity).toBe(expectedAssetQuantity);
-    }
+  describe('tokenMap', () => {
+    it('inserts a single multiasset per asset policy', () => {
+      const multiasset = coreToCsl.tokenMap(txOut.value.assets!);
+      expect(multiasset).toBeInstanceOf(CSL.MultiAsset);
+      expect(multiasset.len()).toBe(3);
```
Comment (mkazlauskas) [inline packages/core/test/CSL/coreToCsl.test.ts:47]: That would be a little clearer as keys of MultiAsset is ScriptHashes which is what we want to test here. Not sure if there's any difference functionally, but I think more explicit is better:

```suggestion
      expect(multiasset.keys().len()).toBe(3);
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/core/test/CSL/coreToCsl.test.ts (hunk 3 above)
Comment (rhyslbw) [reply packages/core/test/CSL/coreToCsl.test.ts]: Done in 37e9d008328ca1f17efc8e93bb8eade91317762f

File: packages/core/test/CSL/testData.ts (hunk 1 above)
Comment (mkazlauskas) [reply packages/core/test/CSL/testData.ts]: Done in [9cb225b](https://github.com/input-output-hk/cardano-js-sdk/pull/274/commits/9cb225be0fea0ec2da65f393f7120be986f0929d)

File: packages/core/src/CSL/coreToCsl/coreToCsl.ts (hunk 2 above)
Comment (mkazlauskas) [reply packages/core/src/CSL/coreToCsl/coreToCsl.ts]: Captured in ADP-1876

**Review by iadmytro — APPROVED** [review]

**Pull request merged**